Launch: Add persistent launch flow button in the block editor

ciampo opened the pull request:
## Changes proposed in this Pull Request

- Adds persistent launch button (as spec'd in pbAok1-1y7-p2 and p9Jlb4-1SN-p2)
- All un-launched sites should now see this button — when clicked, users will be taken to one of two possible launch flows:
   - "Complete setup" if the site was created through Gutenboarding (this flow opens a modal on top of the editor)
   - "Control launch flow" otherwise (this flow redirects the user to a new page)


Related phab patches:
- FSE plugin: D51379-code
- WBE widget: D51381-code



## Testing instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

1. Apply FSE and WBE phab patches to the sandbox (see above for phab patches links)

2. **First flow: a site created via the _control_ flow should display the launch button behind a flag**
   * Create a new site via `/start`
   * Add the newly created site to the sandbox
   * Edit homepage in the calypso block editor — persistent launch button should not appear
   * Add the query param `flags=gutenboarding/persistent-launch-button` to the URL, refresh the editor page
   * "Launch" button should appear
   * Clicking on the launch button should redirect to `/start/launch-site`

3. **Second flow: a site created with the _gutenboarding_ flow should always display the launch button**
   * Create a new site via `/new`
   * Add the newly created site to the sandbox
   * Resize the viewport to a desktop size
   * Edit homepage in the calypso block editor
   * "Launch" button should appear
   * Clicking on the launch button should open the new launch flow in a modal (without redirect)

ciampo commented on apps/editing-toolkit/editing-toolkit-plugin/editor-gutenboarding-launch/index.ts:
```diff
@@ -72,14 +85,23 @@ function updateEditor() {
 
 			recordTracksEvent( 'calypso_newsite_editor_launch_click', {
-				is_new_flow: shouldOpenNewFlow,
+				// TODO: 'is_new_flow' flag — is it still relevant?
+				is_new_flow: shouldOpenNewFlowModal,
```
@razvanpapadopol What changes should we make in terms of analytics for the Persistent launch button?

tjcafferkey commented on apps/editing-toolkit/editing-toolkit-plugin/editor-gutenboarding-launch/index.ts:
```diff
@@ -50,15 +63,15 @@ function updateEditor() {
 		clearInterval( awaitSettingsBar );
 
-		const isMobile = window.innerWidth < 782;
+		const isMobileViewport = window.innerWidth < 782;
```
Magic number. Would be good if we can assign this to a named const somewhere (even if it's in the same file).

ciampo replied:
Good point! @yansern is there a specif reason why you came up with `782`? Just to help to name this variable

yansern replied:
782 is $break-medium from `@wordpress/base-styles/_breakpoints.scss`. 

From package `@automattic/viewport`, there is no such viewpoint width (and also `@automattic/viewport` behaves differently, like a hook? so didn't want to complicate what is just simple JS code here).

yansern commented on apps/editing-toolkit/editing-toolkit-plugin/editor-gutenboarding-launch/index.ts:
```diff
@@ -96,6 +118,10 @@ function updateEditor() {
 		};
 
 		const body = document.querySelector( 'body' );
+		if ( ! body ) {
+			return;
+		}
```
Marco said is for TS fixes.

yansern commented:
Test instructions for first & second flow all works as expected!

razvanpapadopol commented:
Let's hold off from merging this until `gutenboarding/new-launch-mobile` flag is enabled in production so we don't end up having this temporary scenario for Gutenboarding sites:
> Testing for mobile viewports without the "new launch mobile" flag:
> [...]
> - Clicking on the launch button should redirect to /start/launch-site

cc: @yansern

ciampo commented:
`gutenboarding/new-launch-mobile` flag is now enabled in production (see #46807).

I rebased, simplified the test logic, and we added an FSE version bump too.
